src/test/java/com/slack/kaldb/metadata/zookeeper/ZookeeperMetadataStoreImplTest.java

```diff
@@ -496,5 +504,30 @@ public void testFatalErrorHandlerInvocation() throws Exception {
 
     // The FatalErrorHandler is incremented async in a separate thread
     await().until(() -> countingFatalErrorHandler.getCount() == 1);
+
+    // close the underlying zookeeper connection to ensure it's correctly removed
+    closeZookeeperClientConnection();
```

Comment:
Should we add this method in tearDown instead of selectively calling in specific tests?

Reply:
I only added this to the two tests that simulate zookeeper timeouts using the "incorrect" shutdown order, as this is not the desired way to handle the shutdown. If we were to add this to all tests there's a chance that we would be masking shutdown issues given the forcible nature of this approach.